packages/xstate-solid/src/fsm.ts

```diff
@@ -0,0 +1,67 @@
+import type {
+  MachineImplementationsFrom,
+  ServiceFrom,
+  StateFrom,
+  StateMachine
+} from '@xstate/fsm';
+import { createMachine, interpret } from '@xstate/fsm';
+import type { Accessor } from 'solid-js';
+import { createMemo, onCleanup, createComputed } from 'solid-js';
+import { createImmutable } from './createImmutable';
+
+type UseFSMReturn<TService extends StateMachine.AnyService> = [
+  StateFrom<TService>,
+  TService['send'],
+  TService
+];
+
+export function useMachine<TMachine extends StateMachine.AnyMachine>(
+  machine: TMachine,
+  options?: MachineImplementationsFrom<TMachine>
+): UseFSMReturn<ServiceFrom<TMachine>> {
+  const resolvedMachine = createMachine(
+    machine.config,
+    options ? options : (machine as any)._options
+  );
+
+  const service = interpret(resolvedMachine).start();
+
+  return useService(service) as UseFSMReturn<ServiceFrom<TMachine>>;
+}
+
+export function useService<TService extends StateMachine.AnyService>(
+  service: TService | Accessor<TService>
+): UseFSMReturn<TService> {
+  const serviceMemo = createMemo(() =>
+    typeof service === 'function' ? service() : service
+  );
+
+  const initialService = serviceMemo();
+  const [state, setState] = createImmutable(deriveFSMState(initialService));
+
+  createComputed(() => {
```

Comment:
`useActor` used `isInitialActor` variable - what's the difference between this one here without that and that other one with it?

Reply:
`fsm`'s subscribe triggers an update immediately but `useActor`'s subscribe does not. So for `fsm` it synchronizes the state automatically on subscription when the service changes. `useActor` requires setting state using the new actor or our Solid state is out of sync with the actor's actual state